FusionIIIT/applications/hostel_management/models.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+from django.db import models
+from django.contrib.auth.models import User
+from applications.globals.models import ExtraInfo, Staff, Faculty
+from applications.academic_information.models import Student
+from applications.complaint_system.models import Caretaker
+from django.utils import timezone
+
+
+class HostelManagementConstants:
+    ROOM_STATUS = (
+        ('Booked', 'Booked'),
+        ('CheckedIn', 'Checked In'),
+        ('Available', 'Available'),
+        ('UnderMaintenance', 'Under Maintenance'),
+        )
+
+    DAYS_OF_WEEK = (
+            (0, 'Monday'),
+            (1, 'Tuesday'),
+            (2, 'Wednesday'),
+            (3, 'Thursday'),
+            (4, 'Friday'),
+            (5, 'Saturday'),
+            (6, 'Sunday')
+        )
+
+    BOOKING_STATUS = (
+    ("Confirmed" , 'Confirmed'),
+    ("Pending" , 'Pending'),
+    ("Rejected" , 'Rejected'),
+    ("Canceled" , 'Canceled'),
+    ("CancelRequested" , 'Cancel Requested'),
+    ("CheckedIn" , 'Checked In'),
+    ("Complete", 'Complete'),
+    ("Forward", 'Forward')
+    )
+
+
+
+class Hall(models.Model):
+    hall_id = models.CharField(max_length=10)
+    hall_name = models.CharField(max_length=50)
+    max_accomodation = models.IntegerField(default=0)
+    number_students = models.PositiveIntegerField(default=0)
+    hall_caretaker = models.ManyToManyField(Caretaker, on_delete=models.CASCADE)
+    hall_warden = models.ManyToManyField(Faculty, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.hall_id
+    
+
+class GuestRoomDetail(models.Model):
+    hall = models.ForeignKey(Hall, on_delete=models.CASCADE)
+    room_no = models.CharField(max_length=4, unique=True)
+    room_status  = models.CharField(max_length=20, choices=HostelManagementConstants.ROOM_STATUS, default='Available')
+
+    def __str__(self):
+        return self.room_no
+
+
+class GuestRoomBooking(models.Model):
+    hall = models.ForeignKey(Hall, on_delete=models.CASCADE)
+    student_id = models.ForeignKey(Student, on_delete=models.CASCADE)
+    guest_name = models.CharField(max_length=100)
+    guest_phone = models.CharField(max_length=15)
+    guest_email = models.CharField(max_length=40, blank=True)
+    guest_address = models.TextField(blank=True)
+    number_of_rooms =  models.IntegerField(default=1,null=True,blank=True)
+    guest_room_id = models.ManyToManyField(GuestRoomDetail, on_delete=models.CASCADE)
+    total_guest = models.IntegerField(default=1)
+    relation_with_student = models.CharField(max_length=50)
```

Comment:
I think this field is not present in the form, and hence is not required (I think this is because maybe faculty can also book the guest rooms, you need to confirm that from Atul sir).

But if you have a reason, keep it for now but try not to use it until necessary.

Can you also change the field name to `relationship_with_intender`?